### PR TITLE
Simplify CompileMetadata

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/language-service/src/typescript_host.ts [hunk 1]
```diff
@@ -235,12 +235,12 @@ export class TypeScriptServiceHost implements LanguageServiceHost {
       const urlResolver = createOfflineCompileUrlResolver();
       for (const module of ngModuleSummary.ngModules) {
         for (const directive of module.declaredDirectives) {
-          const directiveMetadata =
+          const {metadata, annotation} =
```
Comment (chuckjaz) [inline modules/@angular/language-service/src/typescript_host.ts]: Consider changing this to `const {metadata} =` instead as `annotation` is not used.

File: modules/@angular/language-service/src/language_service.ts [hunk 2]
```diff
@@ -108,7 +108,7 @@ class LanguageServiceImpl implements LanguageService {
     let result: AstResult;
     try {
       const directive =
```
Comment (chuckjaz) [inline modules/@angular/language-service/src/language_service.ts]: Consider changing to `const {metadata} = ` and renaming `directive` to `metadata`. This is more consistent with other calls to `getNonNormalizedDirectiveMetadata()`.

**Comment by chuckjaz** [comment]
LGTM with small updates.

File: modules/@angular/language-service/src/typescript_host.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/language-service/src/typescript_host.ts]: It is used here, see L242

File: modules/@angular/language-service/src/language_service.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/language-service/src/language_service.ts]: Done

**Comment by tbosch** [comment]
Note: This is green on TAP.

**Comment by vsavkin** [comment]
Merged https://github.com/angular/angular/commit/2f7492c986bb64669bfc2d3d6da60f09db13ded7

**Pull request closed without merging**